tensorflow/compiler/xla/service/gpu/gpu_compiler.cc

```diff
@@ -852,7 +852,7 @@ static Status CompileModuleToLlvmIrImpl(
     HloModule* hlo_module, llvm::LLVMContext* llvm_context,
     const std::string& target_triple, const std::string& data_layout,
     const std::string& platform_name, GpuDeviceInfo gpu_device_info,
-    se::CudaComputeCapability cuda_compute_capability,
+    se::CudaComputeCapability cuda_compute_capability, std::string amdgpu_arch,
```

Comment:
Maybe absl::string_view?

Reply:
I guess if we pass CudaComputeCapability it might make sense to pass AMD capabilities as well.

Two questions though:

1) Maybe it's better to have a class instead?
2) What about passing `absl::variant<>` of both? While we're there, can create a wrapping class GpuComputeCapability.

Reply:
@cheshire , are you proposing that we should
* create a `se::RocmComputeCapability` class 
* replace references to `CudaComputeCapability` with references to `absl::variant<RocmComputeCapability, CudaComputeCapability>` instead?

There are a lot of references to `CudaComputeCapability` in the codebase, will all of them need to be updated to use the `absl::variant<...>` ?

Note that having a common `GpuComputeCapability` abstract class to define an interface that is then implemented by concrete `RocmCompuateCapability`/`CudaComputeCapability` class might not work from a design point of view, since the two do not share anything in common. There is `major` / `minor` versioning with `gfx...` HW, and feature support (fast fp16, fast matrix math, etc) is determined on a per `gfx...` type basis.

Reply:
@cheshire ,  please review the change in the following PR (in the ROCm fork)
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/1595

It introduces the `RocmComputeCapability` wrapper that you have proposed here. Once we have it integrated in the ROCm fork, I can update this PR to include those changes

thanks